cmd/lotus-provider/run.go

```diff
@@ -345,3 +253,147 @@ func StorageAuth(apiKey string) (sealer.StorageAuth, error) {
 	headers.Add("Authorization", "Bearer "+string(token))
 	return sealer.StorageAuth(headers), nil
 }
+
+type Deps struct {
+	cfg        *config.LotusProviderConfig
+	db         *harmonydb.DB
+	full       api.FullNode
+	verif      storiface.Verifier
+	lw         *sealer.LocalWorker
+	as         *ctladdr.AddressSelector
+	maddrs     []dtypes.MinerAddress
+	stor       *paths.Remote
+	si         *paths.IndexProxy
+	localStore *paths.Local
+	listenAddr string
+}
+
+func getDeps(ctx context.Context, cctx *cli.Context) (*Deps, error) {
+	// Open repo
+
+	repoPath := cctx.String(FlagRepoPath)
+	fmt.Println("repopath", repoPath)
+	r, err := repo.NewFS(repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	ok, err := r.Exists()
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		if err := r.Init(repo.Provider); err != nil {
+			return nil, err
+		}
+	}
+
+	db, err := makeDB(cctx)
+	if err != nil {
+		return nil, err
+	}
+
+	///////////////////////////////////////////////////////////////////////
+	///// Dependency Setup
+	///////////////////////////////////////////////////////////////////////
+
+	// The config feeds into task runners & their helpers
+	cfg, err := getConfig(cctx, db)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugw("config", "config", cfg)
+
+	var verif storiface.Verifier = ffiwrapper.ProofVerifier
+
+	as, err := provider.AddressSelector(&cfg.Addresses)()
+	if err != nil {
+		return nil, err
+	}
+
+	de, err := journal.ParseDisabledEvents(cfg.Journal.DisabledEvents)
+	if err != nil {
+		return nil, err
+	}
+	j, err := fsjournal.OpenFSJournalPath(cctx.String("journal"), de)
+	if err != nil {
+		return nil, err
+	}
+
+	full, fullCloser, err := cliutil.GetFullNodeAPIV1LotusProvider(cctx, cfg.Apis.ChainApiInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			fullCloser()
+			_ = j.Close()
+		}
```

Comment:
this isn't ideal as it won't give us the correct shutdown order, but I don't think we care about that much in lotus-provider

Reply:
(just a nit, not blocking merge)